src/common.jl

```diff
@@ -72,6 +72,33 @@ nsamples(::Type{D}, x::AbstractMatrix) where {D<:Sampleable{Multivariate}} = siz
 nsamples(::Type{D}, x::Number) where {D<:Sampleable{Matrixvariate}} = 1
 nsamples(::Type{D}, x::Array{Matrix{T}}) where {D<:Sampleable{Matrixvariate},T<:Number} = length(x)
 
+function Base.:(==)(s1::A, s2::B) where {A<:Sampleable, B<:Sampleable}
+    nameof(A) == nameof(B) || return false
+    fields = fieldnames(A)
+    fields == fieldnames(B) || return false
+
+    for f in fields
+        isdefined(s1, f) && isdefined(s2, f) || return false
+        getfield(s1, f) == getfield(s2, f) || return false
+    end
+
+    return true
+end
+
+function Base.isequal(s1::A, s2::B) where {A<:Sampleable, B<:Sampleable}
```

Comment:
is wonder if we should also define `isapprox` the same way (just also allowing `kwargs...`)?
```julia
function Base.isapprox(s1::A, s2::Bl kwargs...) where {A<:Sampleable, B<:Sampleable}
```